### PR TITLE
Mfw 3564 ctid delete

### PR DESCRIPTION
**Pull request by john-phillips-arista**
Pull in new code in support of MFW-3564, to keep up-to-date with upstream.

### CODE REVIEW

File: alignedbuff/alignedbuff.go [hunk 1]
```diff
@@ -118,6 +129,40 @@ func (a *AlignedBuff) Uint64() (uint64, error) {
...
+	len := 0
+	for {
+		if a.data[a.pos+len] == 0x00 {
+			break
+		}
+		len++
+	}
+
+	v := binaryutil.String(a.data[a.pos : a.pos+len])
+	a.pos += len
```
Comment (AlistairJohnParker) [inline alignedbuff/alignedbuff.go:154]: +1 for the '\0' ?

File: alignedbuff/alignedbuff.go [hunk 2]
```diff
@@ -118,6 +129,40 @@ func (a *AlignedBuff) Uint64() (uint64, error) {
...
+func (a *AlignedBuff) String() (string, error) {
+	len := 0
+	for {
+		if a.data[a.pos+len] == 0x00 {
+			break
+		}
+		len++
+	}
+
+	v := binaryutil.String(a.data[a.pos : a.pos+len])
```
Comment (AlistairJohnParker) [inline alignedbuff/alignedbuff.go:153]: presumably this ignores the '\0' at the end ?

File: alignedbuff/alignedbuff.go [hunk 3]
```diff
@@ -190,6 +235,19 @@ func (a *AlignedBuff) PutUint64(v uint64) {
...
+func (a *AlignedBuff) PutInt32(v int32) {
+	a.alignWrite(int32AlignMask)
+	a.data = append(a.data, binaryutil.PutInt32(v)...)
+	a.pos += 4
+}
+
+// PutString marshals a string.
+func (a *AlignedBuff) PutString(v string) {
+	a.data = append(a.data, binaryutil.PutString(v)...)
+	a.pos += len(v)
```
Comment (AlistairJohnParker) [inline alignedbuff/alignedbuff.go:248]: +1 ? Does it put the '\0' at the end ?

File: alignedbuff/alignedbuff.go (hunk 2 above)
Comment (john-phillips-arista) [reply alignedbuff/alignedbuff.go:153]: Dunno, these are upstream changes

**Review by abriles1216 — APPROVED** [review]

**Pull request merged**